### PR TITLE
Add registration page.

### PR DESCRIPTION
**Pull request by DFurnes**
#### What's this PR do?
This pull request adds the web registration flow for Northstar! This allows people to register from [northstar.dosomething.org](https://northstar.dosomething.org) (unlikely), and also means that if you use OpenID Connect to authorize with Phoenix, you can register within that flow and be logged in to the newly created account.

#### How should this be reviewed?
👀 

#### Checklist
- [ ] Documentation added for changed endpoints.
- [ ] Tests added for new features/bug fixes.

---
For review: @angaither @weerd

### CODE REVIEW

**Comment by DFurnes** [comment]
I'll add tests in the morning! 🚥

File: app/Http/Controllers/WebController.php [hunk 1]
```diff
@@ -86,4 +99,43 @@ public function getLogout()
...
+        // "register" to set a new password on their account.
+        $existing = $this->registrar->resolve($request);
+        if ($existing && $existing->hasPassword()) {
+            throw new NorthstarValidationException(['email' => 'A user with that email or mobile has already been registered.']);
+        }
+
+        $this->registrar->validate($request, $existing, [
+            'first_name' => 'required',
+            'birthdate' => 'required|date',
+            'password' => 'required|confirmed|min:6',
```
Comment (angaither) [inline app/Http/Controllers/WebController.php]: birthday and password required?

how will password required work with voting-app?
Also a ~future problem~ is I don't think we can get birthday with facebook auth.

Comment (DFurnes) [inline app/Http/Controllers/WebController.php]: This is only for the web registration flow! The `POST v1/users/` endpoint will still allow us to programmatically register users without either of those.

Comment (DFurnes) [inline app/Http/Controllers/WebController.php]: For Facebook auth, I believe the idea is to provide a "complete your profile" step where we request that you provide the missing fields.

**Pull request merged**